File: problem17.rb
def single_digit
  ary = ["one","two","three","four","five","six","seven","eight","nine"]
end

def teens
  ary = ["eleven","twelve","thirteen","fourteen","fifteen","sixteen","seventeen","eighteen","nineteen"]
end

def tens
  ary = ["ten","twenty","thirty","fourty","fifty","sixty","seventy","eighty","ninety"]
end

def hundreds
  ary = "hundred"
end

def thousands
  ary = "thousand"
end


def convert_to_words(num)
  digits =  num.to_s.split("").map{ |num| num.to_i}
  word = ""
  while digits.length > 0
    if digits.length == 4
      word += single_digit[digits.first - 1] + thousands
      break
    elsif digits.length == 3
      if digits.join("").to_i % 100 == 0
      	word += single_digit[digits.first - 1] + hundreds
      	break
      else
        word += single_digit[digits.first - 1] + hundreds + "and"
      end
    elsif digits.length == 2
      if digits.first == 0
      	digits.delete_at(0)
      	next
      elsif digits.first == 1 
      	if digits.last == 0
          word += tens[digits.first - 1]
        else      		
      	  word += teens[digits.last - 1] 

      	end
      	break
         
      elsif digits.join("").to_i % 10 == 0
      	word += tens[digits.first - 1]
      	break
      else
      	word += tens[digits.first - 1]
      end

    elsif digits.length == 1 
      word += single_digit[digits.first - 1]

    end
   
    digits.delete_at(0)

  end

  word

end


def letter_count
  
  count = 0
  1.upto(1000) do |num| 
    
    convert_to_words(num).chars.each do |digit|
      count += 1
    end
   
   end
  count
end

p letter_count

File: problem17.py
def nums_to_words(n):
  conversion_dict = {
    1: "one",
    2: "two",
    3: "three",
    4: "four",
    5: "five",
    6: "six",
    7: "seven",
    8: "eight",
    9: "nine",
    10: "ten",
    11: "eleven",
    12: "twelve",
    13: "thirteen",
    14: "fourteen",
    15: "fifteen",
    16: "sixteen",
    17: "seventeen",
    18: "eighteen",
    19: "nineteen",
    20: "twenty",
    30: "thirty",
    40: "forty",
    50: "fifty",
    60: "sixty",
    70: "seventy",
    80: "eighty",
    90: "ninety",
    100: "hundred",
    1000: "thousand"
}
  result = ""
  for num,word in reversed(conversion_dict.items()):
    if (n == 0):
      return result
    elif (n <= 10 and n/num > 0):
      return result + word
    elif (n < 100 and n/num > 0):
      if(n%num == 0):
        return result + word
      return result + word + nums_to_words(n%num)
    elif (n/num > 0):
      if (n%num == 0):
        return result + nums_to_words(n/num) + word
      return result + nums_to_words(n/num) + word + "and" + nums_to_words(n%num)

def letter_count(num):
  i = 1
  count = 0
  while (i <=num):
    word = nums_to_words(i)
    count += len(word)
    i += 1
  return count

print letter_count(1000)
